src/server/generic_family.cc

```diff
@@ -588,7 +588,21 @@ OpStatus OpExpire(const OpArgs& op_args, string_view key, const DbSlice::ExpireP
   if (!IsValid(it))
     return OpStatus::KEY_NOTFOUND;
 
-  return db_slice.UpdateExpire(op_args.db_cntx, it, expire_it, params);
+  auto res = db_slice.UpdateExpire(op_args.db_cntx, it, expire_it, params);
+
+  // If the value was deleted, replicate as DEL.
+  // Else, replicate as PEXPIREAT with exact time.
+  if (auto journal = op_args.shard->journal(); journal && res.ok()) {
+    if (res.value() == -1) {
+      op_args.RecordJournal("DEL"sv, ArgSlice{key});
+    } else {
+      auto time = absl::StrCat(res.value());
+      // Note: Don't forget to change this when adding arguments to expire commands.
+      op_args.RecordJournal("PEXPIREAT"sv, ArgSlice{time});
```

Comment:
where is the key in PEXPIREAT journal write?

Reply:
Damn 😓  I'll fix it in the next PR